### PR TITLE
chore: align ButtonIcons with design

### LINKED ISSUE #9880: [MOB] Align ButtonIcons with design

### PR DESCRIPTION
**Pull request by brianacnguyen**
## **Description**
This PR updates the `ButtonIcon` component to align with the `ButtonIcon` [design in Figma](https://www.figma.com/design/HKpPKij9V3TpsyMV1TpV7C/DS-Components?m=auto&node-id=13906-40534&t=FEaqTRh2Fzwbx5UP-1). 
- The `variants` prop has been removed and consolidated with the `iconColorOverride` prop, allowing developers to configure icon colors more efficiently. 
- A `md` size has been added to the size prop. 
- All instances of the ButtonIcon component have been updated to reflect these changes.

## **Related issues**

Fixes: #9880 

## **Manual testing steps**

Reference videos for examples

### **After**
- `ButtonIcon` stories
https://github.com/MetaMask/metamask-mobile/assets/14355083/2021c359-50ad-406a-bd3b-ccd3d4fb44c2

- `Navbar` changes
https://github.com/MetaMask/metamask-mobile/assets/14355083/e0f3abe7-65be-42f6-bd79-04e31604697a

- `Banner` changes
https://github.com/MetaMask/metamask-mobile/assets/14355083/397b5ea2-6aaa-49d8-b953-1df22d73a40b

- `BottomSheetHeader` changes
https://github.com/MetaMask/metamask-mobile/assets/14355083/e19bbab7-1210-4b25-9121-c47529600c68

- `SheetHeader` changes
https://github.com/MetaMask/metamask-mobile/assets/14355083/f94989ea-da30-4f10-a631-025363b66c9d

- `Token` changes
https://github.com/MetaMask/metamask-mobile/assets/14355083/f372f868-ddce-44cb-9a1e-88ba07edd6e4

- `Onboarding` changes
https://github.com/MetaMask/metamask-mobile/assets/14355083/e3d79cb0-d2c8-4efc-a5bd-6036343b3bd2

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: app/component-library/components/Banners/Banner/foundation/BannerBase/BannerBase.constants.tsx [hunk 1]
```diff
@@ -25,10 +22,12 @@ export const DEFAULT_BANNERBASE_TITLE_TEXTVARIANT = TextVariant.BodyLGMedium;
 export const DEFAULT_BANNERBASE_DESCRIPTION_TEXTVARIANT = TextVariant.BodyMD;
 export const DEFAULT_BANNERBASE_ACTIONBUTTON_VARIANT = ButtonVariants.Link;
```
Comment (brianacnguyen) [inline app/component-library/components/Banners/Banner/foundation/BannerBase/BannerBase.constants.tsx:23]: https://github.com/MetaMask/metamask-mobile/assets/14355083/397b5ea2-6aaa-49d8-b953-1df22d73a40b

File: app/component-library/components/BottomSheets/BottomSheetHeader/BottomSheetHeader.tsx [hunk 2]
```diff
@@ -24,15 +24,15 @@ const BottomSheetHeader: React.FC<BottomSheetHeaderProps> = ({
   const startAccessory = onBack && (
     <ButtonIcon
       iconName={IconName.ArrowLeft}
-      variant={ButtonIconVariants.Secondary}
+      iconColor={IconColor.Default}
```
Comment (brianacnguyen) [inline app/component-library/components/BottomSheets/BottomSheetHeader/BottomSheetHeader.tsx:27]: https://github.com/MetaMask/metamask-mobile/assets/14355083/e19bbab7-1210-4b25-9121-c47529600c68

File: app/component-library/components/Buttons/ButtonIcon/ButtonIcon.stories.tsx [hunk 3]
```diff
@@ -15,28 +15,31 @@ const ButtonIconMeta = {
       control: {
         type: 'select',
       },
-      defaultValue: SAMPLE_BUTTONICON_PROPS.iconName,
     },
```
Comment (brianacnguyen) [inline app/component-library/components/Buttons/ButtonIcon/ButtonIcon.stories.tsx:18]: https://github.com/MetaMask/metamask-mobile/assets/14355083/2021c359-50ad-406a-bd3b-ccd3d4fb44c2

File: app/component-library/components/Buttons/ButtonIcon/ButtonIcon.types.ts [hunk 4]
```diff
@@ -19,39 +20,22 @@ export type IconSizeByButtonIconSize = {
...
-   * Size of the buttonIcon.
+   * Optional prop to configure the size of the buttonIcon.
    */
   size?: ButtonIconSizes;
   /**
-   * Icon color to override
+   * Optional prop for the color of the icon.
    */
-  iconColorOverride?: string;
+  iconColor?: string | IconColor;
```
Comment (brianacnguyen) [inline app/component-library/components/Buttons/ButtonIcon/ButtonIcon.types.ts:38]: both `variant` and `iconColorOverride` dictates color, so consolidated into `iconColor`

File: app/component-library/components/Sheet/SheetHeader/SheetHeader.tsx [hunk 5]
```diff
@@ -33,7 +33,7 @@ const SheetHeader = ({
         {onBack && (
           <ButtonIcon
             testID={SHEET_HEADER_BACK_BUTTON_ID}
-            variant={ButtonIconVariants.Secondary}
```
Comment (brianacnguyen) [inline app/component-library/components/Sheet/SheetHeader/SheetHeader.tsx:36]: https://github.com/MetaMask/metamask-mobile/assets/14355083/f94989ea-da30-4f10-a631-025363b66c9d

File: app/components/UI/Navbar/index.js [hunk 6]
```diff
@@ -1002,20 +1000,19 @@ export function getWalletNavbarOptions(
       <View style={styles.leftButtonContainer}>
         {isNotificationsFeatureEnabled() && (
           <ButtonIcon
-            variant={ButtonIconVariants.Primary}
```
Comment (brianacnguyen) [inline app/components/UI/Navbar/index.js:1005]: https://github.com/MetaMask/metamask-mobile/assets/14355083/e0f3abe7-65be-42f6-bd79-04e31604697a

File: app/components/UI/OnboardingWizard/Coachmark/index.js [hunk 7]
```diff
@@ -400,7 +400,7 @@ export default class Coachmark extends PureComponent {
                 iconName={IconName.Arrow2Left}
                 size={IconSize.Sm}
```
Comment (brianacnguyen) [inline app/components/UI/OnboardingWizard/Coachmark/index.js:401]: not sure how to get this from app

File: app/components/UI/Tokens/index.tsx [hunk 8]
```diff
@@ -171,9 +171,8 @@ const Tokens: React.FC<TokensI> = ({ tokens }) => {
           onPressIn={() => {
             setShowScamWarningModal(true);
           }}
-          variant={ButtonIconVariants.Primary}
-          size={IconSize.Lg}
-          iconColorOverride={IconColor.Error}
+          iconColor={IconColor.Error}
```
Comment (brianacnguyen) [inline app/components/UI/Tokens/index.tsx:174]: https://github.com/MetaMask/metamask-mobile/assets/14355083/f372f868-ddce-44cb-9a1e-88ba07edd6e4

**Review by georgewrmarshall — DISMISSED** [review]
LGTM! 
- Checked for instances of removed `iconColorOverride` and `ButtonIconVariants` ✅

File: app/components/UI/OnboardingWizard/Coachmark/index.js (hunk 7 above)
Comment (brianacnguyen) [reply app/components/UI/OnboardingWizard/Coachmark/index.js:401]: not sure how to trigger the onboarding wizard. i've reset the wallet but its still not showing up

Comment (Cal-L) [reply app/components/UI/OnboardingWizard/Coachmark/index.js:401]: You can delete the app and reinstall to prompt it

File: app/components/UI/OnboardingWizard/Coachmark/index.js [hunk 9]
```diff
@@ -400,7 +400,7 @@ export default class Coachmark extends PureComponent {
...
                 onPress={this.onBack}
-                iconColorOverride={colors.primary.inverse}
```
Comment (brianacnguyen) [inline app/components/UI/OnboardingWizard/Coachmark/index.js:403]: https://github.com/MetaMask/metamask-mobile/assets/14355083/e3d79cb0-d2c8-4efc-a5bd-6036343b3bd2

**Review by Cal-L — DISMISSED** [review]
LGTM

**Review by georgewrmarshall — APPROVED** [review]

**Pull request merged**